aws/resource_aws_instance.go

```diff
@@ -1066,6 +1117,35 @@ func InstanceStateRefreshFunc(conn *ec2.EC2, instanceID, failState string) resou
 	}
 }
 
+// VolumeStateRefreshFunc returns a resource.StateRefreshFunc that is used to watch
+// an EC2 root device volume.
+func VolumeStateRefreshFunc(conn *ec2.EC2, volumeID, failState string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := conn.DescribeVolumesModifications(&ec2.DescribeVolumesModificationsInput{
+			VolumeIds: []*string{aws.String(volumeID)},
+		})
+		if err != nil {
+			if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidVolumeID.NotFound" {
```

Comment:
We have a nice helper for these kind of checks - see https://github.com/terraform-providers/terraform-provider-aws/blob/b8d4e1570fc43a2acee6b6e47f63c9db6b067fa2/aws/resource_aws_emr_security_configuration.go#L101